### DISCUSSION TITLE
Ghost in /dynamic_obstacles_map

### DISCUSSION THREAD

**Discussion by s-kajita** [HumanSupportRobot/QandA#98] [Q&A] [answered]
HSRC#56 observes "Ghost" obstacles as shown in the movie. 
It repeats appear and vanish in a few seconds and disturbs the autonomous navigation.
I cleaned LiDAR and Xtion lens but the problem was not solved.

https://github.com/HumanSupportRobot/QandA/assets/10680935/82c9d884-c97c-4f27-9667-cb44ad876771

The topic is /dynamic_obstacle_map published by the node 
/hsrb/head_rgbd_sensor/head_rgbd_sensor_nodelet_manager

How can I solve this problem?

**Comment by hawkina** [comment, marked as answer]
Hey,
at some point we had a similar issue and it turns out that during navigation, the robot takes into account not only the laser scanner but also the rgbd camera. It would see it's hand, think it's an obstacle, and act accordingly. You could test if this is the case for you also by disabling the use of rgbd data for navigation. You can do this by following the answer here: https://qa.hsr.io/en/question/1973/i-want-to-navigate-using-only-the-base_scan/

It tends to be the case that the robot simply sees it's gripper and thinks it is an obstacle, since it does not come with self-filtering for this by default.
If this does not work, could you please share a video of your robots behavior? 
Good luck!

**Reply by s-kajita** [reply]
Great thanks!  I will follow your advice tomorrow.

**Reply by s-kajita** [reply]
It worked！ Thank you very much again!!

**Comment by s-kajita** [comment]
https://github.com/hawkina 's advice did work!

First, I opend dynamic_reconfigure.

         $ hsrb_mode
         HSRB$ rosrun rqt_gui rqt_gui -s reconfigure

Next, I unchecked  /tmc_map_merger/head_rgbd_sensor
![Screenshot from 2024-03-16 17-01-41](https://github.com/HumanSupportRobot/QandA/assets/10680935/95ca64c9-560d-4424-beae-69ec779418fb)

The ghosts (fake obstacles) around HSRC have vanished!

https://github.com/HumanSupportRobot/QandA/assets/10680935/4ed45451-0437-4197-8446-837d90f6ce6f
